src/engine/enginemaster.cpp

```diff
@@ -188,6 +188,9 @@ EngineMaster::EngineMaster(UserSettingsPointer pConfig,
     m_pHeadphoneEnabled = new ControlObject(ConfigKey(group, "headEnabled"));
     m_pHeadphoneEnabled->setReadOnly();
 
+    m_masterBpm = new ControlProxy("[InternalClock]", "bpm", this);
+    m_masterBeatDistance = new ControlProxy("[InternalClock]", "beat_distance", this);
```

Comment:
This feels somewhat hacky to use ControlProxies for this. Couldn't this information be obtained from m_pMasterSync?

Reply:
Updated. For this I had to make the `BaseSyncableListener::masterBpm()` and `BaseSyncableListener::masterBeatDistance()` functions public from protected.